core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -173,6 +181,7 @@ public TenantObject addTrustAnchor(final byte[] publicKey,
         trustedCa.put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey);
         trustedCa.put(TenantConstants.FIELD_PAYLOAD_KEY_ALGORITHM, publicKeyAlgorithm);
         trustedCa.put(TenantConstants.FIELD_AUTO_PROVISIONING_ENABLED, autoProvisioningEnabled);
+        trustedCa.put(TenantConstants.FIELD_AUTO_PROVISION_AS_GATEWAY, autoProvisioningAsGatewayEnabled);
```

Comment:
this property is not relevant for the protocol adapters, is it?
I understand that we use the TenantObject as the vehicle to convey the tenant's configuration within the device registry's services, in particular it is used as the return value of *TenantInformationService.getTenant()*. However, *TenantInformationService* is a strictly registry internal service so to me it feels very strange that we use TenantObject here instead of the *Tenant* object that is used by the TenantManagementService. FMPOV we should keep the information that we expose to the adapters to the required minimum. IMHO we should therefore change *TenantInformationService.getTenant()* to return a *Tenant* instance instead of a *TenantObject* (which is intended to be consumed by adapters). WDYT?

Reply:
At the very first sight I also thought the same seeing that the tenant information returns `TenantObject`. I also think that it should return `Tenant` instance. Currently tenant information service makes use of `TenantService` internally and in my opinion it should be `TenantManagementService`.   I already wanted to to address this in a separate PR as it involves changes in several other places too after finishing #2272.

Reply:
I think it makes more sense to change the `TenantInformationService.getTenant()` to return `Tenant` instance and also to change the `AutowiredTenantInformationService` to make use of the `TenantManagmentService` instead of `TenantService` before further proceeding with this PR.  I will create a PR for that.

Reply:
> @sophokles73 Currently I am working on the refactoring as we discussed above. What I noticed is that the edge way device provisioning uses `EventSender.sendEvent(TenantObject tenant ...)` to send auto-provisioned events. In order to get rid of the `TenantObject` there we need an alternative option for sending events without involving `TenantObject`. For me it seems that starting with change in `EventSender` would slow down the progress of #2272. What I can think of now, is to leave the usage of the `TenantObject` as it is only for the above case.
> 
> That means we need a workaround by converting `Tenant` to `TenantObject` as in the below code only for the sake of `EventSender.sendEvent(TenantObject tenant ...)`. And once the #2272 is over, we could come with a solution to over come that. Or do you have other suggestions?
> 
> ```
>  DeviceRegistryUtils.convertTenant(tenantId, tenant)
>                     .mapTo(TenantObject.class);
> ```

Reply:
Wow, looks like we have opened a can of worms ;-)
I had already been thinking about refactoring the telemetry/event sender from *client-adapter* into a separate module which is not intended to be used by adapters only. As part of that we could introduce a kind of *internal* sender which does not require the information from the `TenantObject`. But I agree that it will be better to postpone that to a subsequent issue/PR.

Reply:
@sophokles73 I have created a PR #2650 for changing the tenant information service to make use of `Tenant` and `TenantManagementService` as we discussed above. Would you mind taking a look?